File: backend/common/src/main/java/fi/hel/allu/common/types/AttachmentType.java
package fi.hel.allu.common.types;

/**
 * List of attachment types.
 */
public enum AttachmentType {
  ADDED_BY_CUSTOMER(false),   // Attachment added by the customer
  ADDED_BY_HANDLER(false),    // Attachment added by the handler (HKR employee)
  DEFAULT(true),              // Application type specific default attachment
  DEFAULT_IMAGE(true),        // Application type specific default image attachment such as picture of traffic arrangements
  DEFAULT_TERMS(true),        // Application type specific default terms attachment
  SUPERVISION(false),         // Supervision attachment
  STATEMENT(false),           // Statement (lausunto) attachment
  OTHER(false);

  private boolean isDefaultAttachment;

  private AttachmentType(boolean isDefaultAttachment) {
    this.isDefaultAttachment = isDefaultAttachment;
  }

  public boolean isDefaultAttachment() {
    return isDefaultAttachment;
  }
}

File: frontend/src/model/application/attachment/attachment-type.js
"use strict";
(function (AttachmentType) {
    AttachmentType[AttachmentType["ADDED_BY_CUSTOMER"] = 0] = "ADDED_BY_CUSTOMER";
    AttachmentType[AttachmentType["ADDED_BY_HANDLER"] = 1] = "ADDED_BY_HANDLER";
    AttachmentType[AttachmentType["DEFAULT"] = 2] = "DEFAULT";
    AttachmentType[AttachmentType["DEFAULT_IMAGE"] = 3] = "DEFAULT_IMAGE";
    AttachmentType[AttachmentType["DEFAULT_TERMS"] = 4] = "DEFAULT_TERMS"; // Hakemustyyppikohtainen ehtoliite
})(exports.AttachmentType || (exports.AttachmentType = {}));
var AttachmentType = exports.AttachmentType;
exports.commonAttachmentTypes = [
    AttachmentType.ADDED_BY_CUSTOMER,
    AttachmentType.ADDED_BY_HANDLER
];
exports.isCommon = function (type) { return exports.commonAttachmentTypes.indexOf(AttachmentType[type]) >= 0; };
